# Introdução

In [2]:
# importando bibliotecas
import math

In [3]:
# calculo do fatorial de cinco

def fatorial(n):
    if n == 0 or n ==1:
        return 1
    resultado = 1
    for i in range(2, n + 1):
        resultado *= i
    return resultado

In [4]:
fatorial(6)

720

In [5]:
# usando a funcao math
math.factorial(6)

720

# Estudos

## 2.1 Estudos Introdução a análise de dados com pandas

In [7]:
import numpy as np
import pandas as pd

In [8]:
series01 = pd.Series([1,3,5,np.nan, 6, 8])

In [9]:
series01.head()

0    1.0
1    3.0
2    5.0
3    NaN
4    6.0
dtype: float64

In [10]:
df = pd.read_csv("/home/isaacnilberto/Home/Isaac/curso_numpy_pandas/universidade_dos_dados/Iris.csv")

In [11]:
df.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [12]:
# criando um dataframe
pessoas = pd.DataFrame({"Nome" : ["Ana", "Carlos"],
                       "Idade": [25,30]})
pessoas

Nome  Idade
0     Ana     25
1  Carlos     30

## 2.2 Análise de dados com pandas

In [48]:
# importando o dado
df01 = pd.read_csv("/home/isaacnilberto/Home/Isaac/curso_numpy_pandas/universidade_dos_dados/insurance.csv")
# fazendo uma copia
df_work = df01.copy()

In [14]:
# verificando as primeiras linhas
df_work.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [15]:
# verificando as ultimas linhas
df_work.tail()

age     sex    bmi  children smoker     region     charges
1333   50    male  30.97         3     no  northwest  10600.5483
1334   18  female  31.92         0     no  northeast   2205.9808
1335   18  female  36.85         0     no  southeast   1629.8335
1336   21  female  25.80         0     no  southwest   2007.9450
1337   61  female  29.07         0    yes  northwest  29141.3603

In [16]:
# verificando o tamanho do data
df_work.shape

(1338, 7)

In [17]:
# verificando as informacoes das colunas
df_work.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [20]:
# acessando os dados
# forma 1
df_work["age"].head()

0    19
1    18
2    28
3    33
4    32
Name: age, dtype: int64

In [22]:
# forma 2
df_work.age.head()

0    19
1    18
2    28
3    33
4    32
Name: age, dtype: int64

In [23]:
# especificando a posicao
df_work.age[3:8]

3    33
4    32
5    31
6    46
7    37
Name: age, dtype: int64

In [24]:
# exibindo a linha 4
df_work.age[4]

32

In [26]:
#acessando varias colunas
df_work[["age","sex","bmi"]].head()

age     sex     bmi
0   19  female  27.900
1   18    male  33.770
2   28    male  33.000
3   33    male  22.705
4   32    male  28.880

In [27]:
# selecionando a primeira linha
df_work.loc[0]

age                19
sex            female
bmi              27.9
children            0
smoker            yes
region      southwest
charges     16884.924
Name: 0, dtype: object

In [28]:
# selecionando a linha 0 e 2
df_work.loc[[0,2]]

age     sex   bmi  children smoker     region    charges
0   19  female  27.9         0    yes  southwest  16884.924
2   28    male  33.0         3     no  southeast   4449.462

In [30]:
# selecionando a linha 0 da coluna bmi
df_work.loc[0,"bmi"]

27.9

In [33]:
# fazendo selecoes
df_work.loc[df_work["age"] > 63]

age         22
sex         22
bmi         22
children    22
smoker      22
region      22
charges     22
dtype: int64

In [36]:
# fazendo multiplas condicoes
df_work.loc[((df_work["age"] > 30) & (df_work["sex"]=="male"))]

age   sex     bmi  children smoker     region      charges
3      33  male  22.705         0     no  northwest  21984.47061
4      32  male  28.880         0     no  northwest   3866.85520
8      37  male  29.830         2     no  northeast   6406.41070
18     56  male  40.300         0     no  southwest  10602.38500
24     37  male  28.025         2     no  northwest   6203.90175
...   ...   ...     ...       ...    ...        ...          ...
1324   31  male  25.935         1     no  northwest   4239.89265
1325   61  male  33.535         0     no  northeast  13143.33665
1327   51  male  30.030         1     no  southeast   9377.90470
1329   52  male  38.600         2     no  southwest  10325.20600
1333   50  male  30.970         3     no  northwest  10600.54830

[446 rows x 7 columns]

In [37]:
# iloc baseia-se no indice
df_work.iloc[0]

age                19
sex            female
bmi              27.9
children            0
smoker            yes
region      southwest
charges     16884.924
Name: 0, dtype: object

In [38]:
df_work.iloc[[3,10]]

age   sex     bmi  children smoker     region      charges
3    33  male  22.705         0     no  northwest  21984.47061
10   25  male  26.220         0     no  northeast   2721.32080

# Manipulação de dados

In [39]:
df_work["Year_Analysis"] = 2023


In [49]:
df_work["abusive"] = np.where(
    df_work["charges"] > 20000,
    "yes",
    "no"
)
df_work.head()

age     sex     bmi  children smoker     region      charges abusive
0   19  female  27.900         0    yes  southwest  16884.92400      no
1   18    male  33.770         1     no  southeast   1725.55230      no
2   28    male  33.000         3     no  southeast   4449.46200      no
3   33    male  22.705         0     no  northwest  21984.47061     yes
4   32    male  28.880         0     no  northwest   3866.85520      no

In [43]:
# excluindo colunas
df_work = df_work.drop(columns="Year_Analysis")
df_work.head()

age     sex     bmi  children smoker     region      charges Abusive
0   19  female  27.900         0    yes  southwest  16884.92400      No
1   18    male  33.770         1     no  southeast   1725.55230      No
2   28    male  33.000         3     no  southeast   4449.46200      No
3   33    male  22.705         0     no  northwest  21984.47061     Yes
4   32    male  28.880         0     no  northwest   3866.85520      No

In [50]:
# trocando yes to 1 and no to 0
df_work["abusive"] = df_work["abusive"].map({
    "yes":1,
    "no":0
})

In [51]:
df_work

age     sex     bmi  children smoker     region      charges  abusive
0      19  female  27.900         0    yes  southwest  16884.92400        0
1      18    male  33.770         1     no  southeast   1725.55230        0
2      28    male  33.000         3     no  southeast   4449.46200        0
3      33    male  22.705         0     no  northwest  21984.47061        1
4      32    male  28.880         0     no  northwest   3866.85520        0
...   ...     ...     ...       ...    ...        ...          ...      ...
1333   50    male  30.970         3     no  northwest  10600.54830        0
1334   18  female  31.920         0     no  northeast   2205.98080        0
1335   18  female  36.850         0     no  southeast   1629.83350        0
1336   21  female  25.800         0     no  southwest   2007.94500        0
1337   61  female  29.070         0    yes  northwest  29141.36030        1

[1338 rows x 8 columns]

In [52]:
# trocando yes to 1 and no to 0
df_work["smoker"] = df_work["smoker"].map({
    "yes":1,
    "no":0
})

In [53]:
df_work.head()

age     sex     bmi  children  smoker     region      charges  abusive
0   19  female  27.900         0       1  southwest  16884.92400        0
1   18    male  33.770         1       0  southeast   1725.55230        0
2   28    male  33.000         3       0  southeast   4449.46200        0
3   33    male  22.705         0       0  northwest  21984.47061        1
4   32    male  28.880         0       0  northwest   3866.85520        0

In [54]:
df_work.describe()

age          bmi     children       smoker       charges  \
count  1338.000000  1338.000000  1338.000000  1338.000000   1338.000000   
mean     39.207025    30.663397     1.094918     0.204783  13270.422265   
std      14.049960     6.098187     1.205493     0.403694  12110.011237   
min      18.000000    15.960000     0.000000     0.000000   1121.873900   
25%      27.000000    26.296250     0.000000     0.000000   4740.287150   
50%      39.000000    30.400000     1.000000     0.000000   9382.033000   
75%      51.000000    34.693750     2.000000     0.000000  16639.912515   
max      64.000000    53.130000     5.000000     1.000000  63770.428010   

           abusive  
count  1338.000000  
mean      0.204036  
std       0.403146  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000

In [55]:
df_work["sex"].value_counts()

sex
male      676
female    662
Name: count, dtype: int64

In [56]:
df_work["sex"].value_counts(normalize=True)

sex
male      0.505232
female    0.494768
Name: proportion, dtype: float64

In [57]:
# agrupando os dados
df_work.groupby("sex")["age"].mean()

sex
female    39.503021
male      38.917160
Name: age, dtype: float64

In [58]:
df_work.groupby("smoker")[["age","charges"]].mean()

age       charges
smoker                         
0       39.385338   8434.268298
1       38.514599  32050.231832

In [59]:
df_work.groupby("smoker")[["age","charges"]].std()

age       charges
smoker                         
0       14.083410   5993.781819
1       13.923186  11541.547176

In [60]:
df_work.groupby(["smoker","sex"])[["charges"]].std()

charges
smoker sex                 
0      female   6060.775970
       male     5908.108989
1      female  11907.536381
       male    11202.670862

# Exercícios

In [61]:
df_work.head()

age     sex     bmi  children  smoker     region      charges  abusive
0   19  female  27.900         0       1  southwest  16884.92400        0
1   18    male  33.770         1       0  southeast   1725.55230        0
2   28    male  33.000         3       0  southeast   4449.46200        0
3   33    male  22.705         0       0  northwest  21984.47061        1
4   32    male  28.880         0       0  northwest   3866.85520        0

In [62]:
df_work.tail()

age     sex    bmi  children  smoker     region     charges  abusive
1333   50    male  30.97         3       0  northwest  10600.5483        0
1334   18  female  31.92         0       0  northeast   2205.9808        0
1335   18  female  36.85         0       0  southeast   1629.8335        0
1336   21  female  25.80         0       0  southwest   2007.9450        0
1337   61  female  29.07         0       1  northwest  29141.3603        1

In [63]:
df_work.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
abusive     0
dtype: int64

In [64]:
df_work.describe()

age          bmi     children       smoker       charges  \
count  1338.000000  1338.000000  1338.000000  1338.000000   1338.000000   
mean     39.207025    30.663397     1.094918     0.204783  13270.422265   
std      14.049960     6.098187     1.205493     0.403694  12110.011237   
min      18.000000    15.960000     0.000000     0.000000   1121.873900   
25%      27.000000    26.296250     0.000000     0.000000   4740.287150   
50%      39.000000    30.400000     1.000000     0.000000   9382.033000   
75%      51.000000    34.693750     2.000000     0.000000  16639.912515   
max      64.000000    53.130000     5.000000     1.000000  63770.428010   

           abusive  
count  1338.000000  
mean      0.204036  
std       0.403146  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000

In [65]:
df_work.smoker.value_counts()

smoker
0    1064
1     274
Name: count, dtype: int64

In [67]:
df_work.groupby("smoker")["charges"].mean()

smoker
0     8434.268298
1    32050.231832
Name: charges, dtype: float64

In [69]:
df_work["bmi"].mean()

30.66339686098655

In [72]:
df_work.groupby("smoker")["bmi"].mean()

smoker
0    30.651795
1    30.708449
Name: bmi, dtype: float64

In [73]:
df_work.groupby("children")["charges"].mean()

children
0    12365.975602
1    12731.171832
2    15073.563734
3    15355.318367
4    13850.656311
5     8786.035247
Name: charges, dtype: float64

In [98]:
df_work["bmi_male"] = np.where( 
                                df_work["sex"] == "male",
                                np.where(df_work["bmi"] < 20, "abaixo do normal",
                                   np.where((df_work["bmi"] >= 20) & (df_work["bmi"] <= 24.9), "normal",
                                            "acima do normal")
                               ),np.nan
)

In [99]:
df_work[df_work["bmi_male"] == "acima do normal"]

age   sex     bmi  children  smoker     region      charges  abusive  \
1      18  male  33.770         1       0  southeast   1725.55230        0   
2      28  male  33.000         3       0  southeast   4449.46200        0   
4      32  male  28.880         0       0  northwest   3866.85520        0   
8      37  male  29.830         2       0  northeast   6406.41070        0   
10     25  male  26.220         0       0  northeast   2721.32080        0   
...   ...   ...     ...       ...     ...        ...          ...      ...   
1324   31  male  25.935         1       0  northwest   4239.89265        0   
1325   61  male  33.535         0       0  northeast  13143.33665        0   
1327   51  male  30.030         1       0  southeast   9377.90470        0   
1329   52  male  38.600         2       0  southwest  10325.20600        0   
1333   50  male  30.970         3       0  northwest  10600.54830        0   

             bmi_male  
1     acima do normal  
2     acima do normal  
4     acima do normal  
8     acima do normal  
10    acima do normal  
...               ...  
1324  acima do normal  
1325  acima do normal  
1327  acima do normal  
1329  acima do normal  
1333  acima do normal  

[562 rows x 9 columns]

In [106]:
df_work[(df_work["bmi"]>24.9) & (df_work["sex"]=="male")].select_dtypes(include=[np.number]).sum()

age         2.216200e+04
bmi         1.838405e+04
children    6.300000e+02
smoker      1.360000e+02
charges     8.353959e+06
abusive     1.440000e+02
dtype: float64

In [100]:
df_work.groupby(["bmi_male"])["smoker"].sum()

bmi_male
abaixo do normal      4
acima do normal     136
nan                 115
normal               19
Name: smoker, dtype: int64

In [127]:
df_work[(df_work["smoker"]==1) &(df_work["bmi"]>24.9) & (df_work["sex"]=="male") & (df_work["bmi_male"]=="acima do normal")]

age   sex     bmi  children  smoker     region      charges  abusive  \
14     27  male  42.130         0       1  southeast  39611.75770        1   
19     30  male  35.300         0       1  southwest  36837.46700        1   
29     31  male  36.300         2       1  southwest  38711.00000        1   
30     22  male  35.600         0       1  southwest  35585.57600        1   
34     28  male  36.400         1       1  southwest  51194.55914        1   
...   ...   ...     ...       ...     ...        ...          ...      ...   
1300   45  male  30.360         0       1  southeast  62592.87309        1   
1301   62  male  30.875         3       1  northwest  46718.16325        1   
1303   43  male  27.800         0       1  southwest  37829.72420        1   
1307   32  male  28.120         4       1  northwest  21472.47880        1   
1321   62  male  26.695         0       1  northeast  28101.33305        1   

             bmi_male  
14    acima do normal  
19    acima do normal  
29    acima do normal  
30    acima do normal  
34    acima do normal  
...               ...  
1300  acima do normal  
1301  acima do normal  
1303  acima do normal  
1307  acima do normal  
1321  acima do normal  

[136 rows x 9 columns]

In [118]:
df_filter.shape[0]

562

In [125]:
df

age   sex     bmi  children  smoker     region      charges  abusive  \
1      18  male  33.770         1       0  southeast   1725.55230        0   
2      28  male  33.000         3       0  southeast   4449.46200        0   
4      32  male  28.880         0       0  northwest   3866.85520        0   
8      37  male  29.830         2       0  northeast   6406.41070        0   
10     25  male  26.220         0       0  northeast   2721.32080        0   
...   ...   ...     ...       ...     ...        ...          ...      ...   
1324   31  male  25.935         1       0  northwest   4239.89265        0   
1325   61  male  33.535         0       0  northeast  13143.33665        0   
1327   51  male  30.030         1       0  southeast   9377.90470        0   
1329   52  male  38.600         2       0  southwest  10325.20600        0   
1333   50  male  30.970         3       0  northwest  10600.54830        0   

             bmi_male  
1     acima do normal  
2     acima do normal  
4     acima do normal  
8     acima do normal  
10    acima do normal  
...               ...  
1324  acima do normal  
1325  acima do normal  
1327  acima do normal  
1329  acima do normal  
1333  acima do normal  

[562 rows x 9 columns]

# Tarefa 2

In [3]:
# importanfdo bibliotecas
import pandas as pd
import numpy as np

In [22]:
df_amazon = pd.read_csv("/home/isaacnilberto/Home/Isaac/curso_numpy_pandas/universidade_dos_dados/amazon.csv", sep = ",", encoding= "latin-1")

In [23]:
print(f"O numero de linhas {df_amazon.shape[0]} e colunas {df_amazon.shape[1]}.")

O numero de linhas 6454 e colunas 5.


In [25]:
df_amazon.head()

year state    month  number        date
0  1998  Acre  Janeiro     0.0  1998-01-01
1  1999  Acre  Janeiro     0.0  1999-01-01
2  2000  Acre  Janeiro     0.0  2000-01-01
3  2001  Acre  Janeiro     0.0  2001-01-01
4  2002  Acre  Janeiro     0.0  2002-01-01

In [26]:
# a diferanca entre a sintaxe .loc e .iloc se da pois uma 
# pega os indices .iloc e a .loc localiza pelos inputs colocados
df_amazon.loc[100:120,["number", "date"]]

number        date
100     3.0  1998-01-01
101     0.0  1999-01-01
102     1.0  2000-01-01
103     1.0  2001-01-01
104     0.0  2002-01-01
105     0.0  2003-01-01
106     5.0  2004-01-01
107    27.0  2005-01-01
108     1.0  2006-01-01
109     4.0  2007-01-01
110     0.0  2008-01-01
111     0.0  2009-01-01
112     1.0  2010-01-01
113    10.0  2011-01-01
114     7.0  2012-01-01
115     8.0  2013-01-01
116    17.0  2014-01-01
117    40.0  2015-01-01
118    87.0  2016-01-01
119    45.0  2017-01-01
120    37.0  1998-01-01

In [29]:
df_amazon.groupby("year")["number"].sum()

year
1998    20013.971
1999    26882.821
2000    27351.251
2001    29071.612
2002    37390.600
2003    42760.674
2004    38453.163
2005    35004.965
2006    33832.161
2007    33037.413
2008    29378.964
2009    39117.178
2010    37037.449
2011    34633.545
2012    40084.860
2013    35146.118
2014    39621.183
2015    41208.292
2016    42212.229
2017    36685.624
Name: number, dtype: float64

In [35]:
df_amazon["state"].unique()

array(['Acre', 'Alagoas', 'Amapa', 'Amazonas', 'Bahia', 'Ceara',
       'Distrito Federal', 'Espirito Santo', 'Goias', 'Maranhao',
       'Mato Grosso', 'Minas Gerais', 'Pará', 'Paraiba', 'Pernambuco',
       'Piau', 'Rio', 'Rondonia', 'Roraima', 'Santa Catarina',
       'Sao Paulo', 'Sergipe', 'Tocantins'], dtype=object)

No ano de 2010 tivemos um total de 37037.449 incêndios.

In [39]:
# fazendo uma analise de incendios por estado
df_amazon_filter = df_amazon[(df_amazon["year"]==2010) & (df_amazon["state"]=="Mato Grosso")]

In [41]:
df_amazon_filter

year        state      month   number        date
2403  2010  Mato Grosso    Janeiro  298.000  2010-01-01
2423  2010  Mato Grosso  Fevereiro  431.000  2010-01-01
2443  2010  Mato Grosso      Março  605.000  2010-01-01
2463  2010  Mato Grosso      Abril  769.000  2010-01-01
2483  2010  Mato Grosso       Maio    1.313  2010-01-01
2503  2010  Mato Grosso      Junho    1.025  2010-01-01
2523  2010  Mato Grosso      Julho    3.442  2010-01-01
2543  2010  Mato Grosso     Agosto   14.608  2010-01-01
2563  2010  Mato Grosso   Setembro   18.366  2010-01-01
2583  2010  Mato Grosso    Outubro    4.465  2010-01-01
2603  2010  Mato Grosso   Novembro  768.000  2010-01-01
2623  2010  Mato Grosso   Dezembro  846.000  2010-01-01
2642  2010  Mato Grosso    Janeiro   72.000  2010-01-01
2662  2010  Mato Grosso  Fevereiro  157.000  2010-01-01
2682  2010  Mato Grosso      Março  149.000  2010-01-01
2702  2010  Mato Grosso      Abril  136.000  2010-01-01
2722  2010  Mato Grosso       Maio  105.000  2010-01-01
2742  2010  Mato Grosso      Junho  149.000  2010-01-01
2762  2010  Mato Grosso      Julho  589.000  2010-01-01
2782  2010  Mato Grosso     Agosto    1.721  2010-01-01
2802  2010  Mato Grosso   Setembro    1.505  2010-01-01
2822  2010  Mato Grosso    Outubro  597.000  2010-01-01
2842  2010  Mato Grosso   Novembro  259.000  2010-01-01
2862  2010  Mato Grosso   Dezembro  276.000  2010-01-01

In [40]:
df_amazon_filter["number"].sum()

6252.445

Temos um total de 6252.445 incêndios no Mato Grosso.

In [42]:
df_amazon_filter2 = df_amazon[(df_amazon["year"]==2010) & (df_amazon["state"]=="Acre")]

In [44]:
df_amazon_filter2["number"].mean()

39.601083333333335

In [45]:
df_amazon_filter2["number"].std()

87.35678433918761

Temos uma media de incêndios de39.601083333333335, um desvio padrão de 87.35678433918761 no Estado do Acre.

In [48]:
df_amazon.groupby("state")["number"].mean().sort_values()

state
Sergipe              13.543933
Distrito Federal     14.899582
Alagoas              19.350000
Espirito Santo       27.389121
Rio                  62.985865
Acre                 77.255356
Rondonia             84.876272
Amapa                91.345506
Santa Catarina      101.924067
Roraima             102.029598
Pernambuco          102.502092
Pará                102.561272
Maranhao            105.142808
Paraiba             109.698573
Ceara               127.314071
Amazonas            128.243218
Tocantins           141.037176
Minas Gerais        156.800243
Goias               157.721841
Piau                158.174674
Bahia               187.222703
Mato Grosso         201.351523
Sao Paulo           213.896226
Name: number, dtype: float64

A UF que possui o maior numero de incêndios é o Estado de SP e o que possui o menor índice é o Estado do Sergipe.

In [49]:
df_amazon

year      state     month  number        date
0     1998       Acre   Janeiro     0.0  1998-01-01
1     1999       Acre   Janeiro     0.0  1999-01-01
2     2000       Acre   Janeiro     0.0  2000-01-01
3     2001       Acre   Janeiro     0.0  2001-01-01
4     2002       Acre   Janeiro     0.0  2002-01-01
...    ...        ...       ...     ...         ...
6449  2012  Tocantins  Dezembro   128.0  2012-01-01
6450  2013  Tocantins  Dezembro    85.0  2013-01-01
6451  2014  Tocantins  Dezembro   223.0  2014-01-01
6452  2015  Tocantins  Dezembro   373.0  2015-01-01
6453  2016  Tocantins  Dezembro   119.0  2016-01-01

[6454 rows x 5 columns]

In [55]:
estados_sudeste = ["Sao Paulo", "Rio de Janeiro", "Espirito Santo", "Minas Gerais"]
estados_centro_oeste = ["Goias", "Mato Grosso", "Distrito Federal"]

df_amazon["regiao_sudeste"]= np.where(df_amazon["state"].isin(estados_sudeste), 1, 0)
df_amazon["regiao_centro_oeste"]= np.where(df_amazon["state"].isin(estados_centro_oeste), 1, 0)

In [56]:
df_amazon

year      state     month  number        date  regiao_sudeste  \
0     1998       Acre   Janeiro     0.0  1998-01-01               0   
1     1999       Acre   Janeiro     0.0  1999-01-01               0   
2     2000       Acre   Janeiro     0.0  2000-01-01               0   
3     2001       Acre   Janeiro     0.0  2001-01-01               0   
4     2002       Acre   Janeiro     0.0  2002-01-01               0   
...    ...        ...       ...     ...         ...             ...   
6449  2012  Tocantins  Dezembro   128.0  2012-01-01               0   
6450  2013  Tocantins  Dezembro    85.0  2013-01-01               0   
6451  2014  Tocantins  Dezembro   223.0  2014-01-01               0   
6452  2015  Tocantins  Dezembro   373.0  2015-01-01               0   
6453  2016  Tocantins  Dezembro   119.0  2016-01-01               0   

      regiao_centro_oeste  
0                       0  
1                       0  
2                       0  
3                       0  
4                       0  
...                   ...  
6449                    0  
6450                    0  
6451                    0  
6452                    0  
6453                    0  

[6454 rows x 7 columns]

In [62]:
print(f" A diferenca entre a regiao centro oeste e sudeste é: {df_amazon['regiao_centro_oeste'].sum() - df_amazon['regiao_sudeste'].sum()}") 

 A diferenca entre a regiao centro oeste e sudeste é: 239
